### PR TITLE
internationalization in django

### LINKED ISSUE #941: Internationalization in Django
### Brief Summary: 
Django is one of the most popular web frameworks known for its battery-inclusion. One of the inbuilt features of Django is internationalization (also known as i18n). This article aims to introduce readers to internationalization. This article shows how to utilize i18n to reduce language barriers in web applications by offering their content in languages tailored to the audience.

### Key Takeaways: 
Readers should be comfortable with Django.
- What is internationalization and why it is needed.
- Understanding Translation strings and message files
- Pluralization
- Language file creations
-  Using translations in templates

### References: 
Django Documentation - https://docs.djangoproject.com/en/3.1/topics/i18n/

### PR DESCRIPTION
**Pull request by Damephena**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #941 issue.

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by lalith1403** [comment]
@Damephena Welcome to the eng-ed program. The article is fascinating. Congratulations on a well-written article. I will check if the code works fine in a couple of days. Meanwhile, include a hero image, a headshot pic, and a short introductory bio for creating your profile. For more details, refer to the contributing guidelines.

**Comment by Damephena** [comment]
Thank you @lalith1403. I wasn't sure if I needed to include the photo credit for image gotten from Unsplash that's why I didn't upload the hero image.

I will update the PR with the items stated.

**Comment by hectorkambow** [comment]
Plagiarism check done. 

@Damephena  👋  
Thank you for submitting your 1st article to the EngEd program and welcome. 

Just wanted to note that your article raise some very minor flags on our 3rd party plagiarism checker (see attachement) 
Please be sure we are adding proper sources within the article wherever applicable. 

Thank you 👍 

[Internalization in Django.pdf](https://github.com/section-io/engineering-education/files/5798874/Internalization.in.Django.pdf)

File: articles/internationalization-in-django/index.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    return HttpResponse(page)
+```
+
+To use gettext() on Windows, you will need to download the executable file from [here](https://mlocati.github.io/articles/gettext-iconv-windows.html)
+
+Django comes with the `django-admin makemessages` command, which creates and updates the message files by pulling out all strings marked for translation. Run the command to create a message file for our simple app:
+
+If you execute `django-admin makemessages --all` and run into CommandError like below, make sure you have gettext binary file installed. `de` and `fr` are the language codes for German and French, respectively.
+
+```bash
```
Comment (lalith1403) [inline articles/internationalization-in-django/index.md:136]: Check the ' symbol inside the bash code block. It is ending abruptly.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
I found an error in your code. I have rectified it. To be safe, can you recheck if the code works. Double-check on your end, and let me know. 

I will review it again after you make the changes to avoid plagiarism. Ensuring minimal errors in such huge codebases ensures faster publishing.

File: articles/internationalization-in-django/index.md (hunk 1 above)
Comment (lalith1403) [reply articles/internationalization-in-django/index.md:137]: I think it can be represented as an escape sequence, \'.

**Comment by Damephena** [comment]
Hello @hectorkambow, I read the plagiarism document generated. I included Django documentation as a reference since I quoted directly from the Django Project.

A few things such as the code block which contains translation file content was flagged. Unfortunately, I cannot replace that considering it comes from the Django generated translation (.mo and .po).

Thank you.

**Review by lalith1403 — APPROVED** [review]
@Damephena I learned a lot about internalization. Very detailed article. Well done. :)

**Comment by lalith1403** [comment]
@Damephena Before we proceed with the final round of review, we would need your headshot pic in the same folder as the description. Along with the headshot pic, a hero image used as the thumbnail image for your article is needed. You can download any 600*360 image from [unsplash](unsplash.com) and store it in the same folder as the article. 

For now, you may use this [image](https://unsplash.com/photos/cwqG1N1AtI0),

**Comment by Damephena** [comment]
Thank you @lalith1403 :) 
I have included a headshot as well as a resized hero image of 600 x 360 jpeg.

**Pull request merged**